python/lsst/analysis/tools/tasks/metadataAnalysis.py

```diff
@@ -89,3 +106,29 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
             raise NoWorkFound(f"No metadata entries for {taskName}.")
         outputs = self.run(data=metadata, plotInfo=plotInfo)
         butlerQC.put(outputs, outputRefs)
+
+
+class DictTypeDatasetMetadataAnalysisTask(AnalysisPipelineTask):
```

Comment:
Should we simply call this `DatasetMetadataAnalysisTask`, or do we expect there to be other, non-dict datasets?

Reply:
We also expect non-dict metadata that simply offer a value for each metric name; I can add one for such regular `1:1` mappings for completeness. This specific ticket focuses on the interactions between many amplifiers on a detector, so it made sense to store information about all those so-called _interface_ offsets in a dictionary that pairs amp names with the offsets, rather than cluttering our metadata with less human-readable names.

Reply:
Introduced the `DatasetMetadataAnalysisTask` class with the `metricsStoredAsDict` config option to make it more generic.

Reply:
In  that case, do feel free to revert back to `DictTypeDatasetMetadataAnalysisTask` and remove the `metricsStoredAsDict` parameter. I'll leave that up to you, though.

Reply:
I think having it this way with`metricsStoredAsDict` is more future-proof because most metrics I know are not stored as dictionaries.